### ISSUE TITLE
Panic when parsing responses with missing reason phrases

### ISSUE THREAD

**Issue by kpcyrd** [hyperium/hyper#1587]
I found a server that is sending replies that trigger a panic in the tokio worker:

```
DEBUG 2018-06-28T12:18:03Z: tokio_core::reactor: loop process, Duration { secs: 0, nanos: 128720 }
DEBUG 2018-06-28T12:18:03Z: hyper::proto::h1::io: flushed 57 bytes
DEBUG 2018-06-28T12:18:03Z: tokio_reactor: loop process - 1 events, 0.000s
DEBUG 2018-06-28T12:18:03Z: hyper::proto::h1::io: read 488 bytes
thread 'tokio-runtime-worker-3' panicked at 'header name already validated: InvalidHeaderName { _priv: () }', libcore/result.rs:945:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.
DEBUG 2018-06-28T12:18:03Z: tokio_reactor: dropping I/O source: 0
```

This happened reliably for every response, so I've sniffed the reply (some infos redacted).

```
HTTP/1.1 302\r\n
Server:\r\n
Date: Thu, 28 Jun 2018 12:18:03 GMT\r\n
Cache-Control: no-cache,no-store,must-revalidate,post-check=0,pre-check=0\r\n
Location: https://xxx-redacted\r\n
Connection: close\r\n
\r\n
```

It seems `Server:\r\n` in the header section may be causing issues.

**Comment by seanmonstar** [comment]
Yikes, definitely a bug! Thanks for pointing this out!

I'm currently thinking that in the parse code, headers that have an empty value should just be dropped, as there's no real value in it...

**Comment by seanmonstar** [comment]
Setup a test case for this, and it turns out to not be the problem. It's actually the missing reason code. It's thinking a header name is `302\r\nServer`...

**Comment by seanmonstar** [comment]
The parse bug has been fixed in httparse, released as 1.3.2. A cargo update will bring the fix in for anyone!
